lib/logger/lib/logger.ex

```diff
@@ -592,4 +592,26 @@ defmodule Logger do
 
   defp notify(:sync, msg),  do: GenEvent.sync_notify(Logger, msg)
   defp notify(:async, msg), do: GenEvent.notify(Logger, msg)
+
+  defp handle_unused_variable_warnings(data) do
+    # We collect all the ASTs of variables (leaving `data` unchanged).
+    {^data, variables} = Macro.postwalk(data, [], fn
+      {name, _meta, scope} = var, acc when is_atom(name) and is_atom(scope) ->
+        {var, [var|acc]}
```

Comment:
What if we store it like `{name, var}` here to have better pattern in `for`? Also spaces around `|`. :bowtie: